internal/pkg/http/client/pool_test.go

```diff
@@ -230,6 +227,9 @@ func TestErrorInSubRequest_ParallelRequests(t *testing.T) {
 		logger.Info("Request2 processed.")
 	})
 
+	// Request 1 wait for request 2
+	request1.WaitFor(request2)
+
```

Comment:
Problem bol priamo v teste. `WaitFor` garantuje poradie, pre vsetky callbacks, ktore uz su definovane.

Reply:
Kedze tak je implementovane `WaitFor`.

https://github.com/keboola/keboola-as-code/blob/d79671f10a86fe69ad08b9291169b019e798e2a0/internal/pkg/http/client/request.go#L183-L192

Tj. k existujucim `listeners` v `subRequest` prida dalsi, a ten odomkne hlavny request (pomocou `waitGroup`).

Reply:
AK sa do `sub-request` pridaju dalsie `listeners` po volani `WaitFor`, tak tych sa to uz nijak netýka (kedze sa pridaju za toto ^^^ `OnResponse`).